main.tf

```
@@ -16,41 +22,53 @@ locals {
 }
 
 resource "aws_security_group" "this" {
-  count = var.create_vpc_endpoints ? 1 : 0
+  count = var.create_sg_per_endpoint ? length(var.vpc_endpoint_services) : 1
```

This should still consider `var.create_vpc_endpoints`, as that toggle is used to disable the entire module.
```suggestion
  count = var.create_vpc_endpoints ? (var.create_sg_per_endpoint ? length(var.vpc_endpoint_services) : 1) : 0
```